Fix 53 TypeScript compilation errors across codebase

## Summary
`npm run typecheck` produces 53 errors across multiple files. This blocks CI type checking and developer experience.

## Key Error Categories

### 1. Renderer `window.api` type errors (SettingsPanel.tsx, SubtitleOverlay.tsx, App.tsx)
- `Property 'api' does not exist on type 'Window & typeof globalThis'`
- `Cannot find namespace 'JSX'`
- `'result' is of type 'unknown'`

### 2. `ws` module types missing (ws-audio-server.ts)
- `Cannot find module 'ws' or its corresponding type declarations`
- Implicit `any` on error parameters

### 3. Type mismatches in core files
- `model-downloader.ts:355` — Buffer type incompatibility in hash stream
- `GeminiTranslator.ts:7` — Incomplete language record (missing 10+ languages)
- `main/index.ts:34,564` — AppSettings ↔ Record<string, unknown> conversion errors
- `slm-worker.ts:286,373` — `repeatPenalty: number` vs `LlamaChatSessionRepeatPenalty`

### 4. Benchmark/E2E config issues
- `benchmark/` — `import.meta` requires es2020+ module setting
- `benchmark/src/report.ts` — Set iteration requires `downlevelIteration`
- `e2e/`, `playwright.config.ts` — Missing `@playwright/test` type declarations

## Suggested Fix
1. Add `@types/ws` to devDependencies and verify installation
2. Augment `Window` interface in renderer with `api` property types
3. Fix type mismatches in GeminiTranslator, model-downloader, slm-worker
4. Add separate tsconfig for benchmark/ and e2e/ directories

--- a/.gitignore
+++ b/.gitignore
@@ -8,3 +8,4 @@ logs/
 src/renderer/public/vad/
 benchmark/results/
 benchmark/models/
+*.tsbuildinfo

--- a/src/engines/model-downloader.ts
+++ b/src/engines/model-downloader.ts
@@ -352,7 +352,7 @@ async function doDownloadWithResume(
         const hash = await new Promise<string>((resolve, reject) => {
           const hasher = createHash('sha256')
           const stream = createReadStream(partialPath)
-          stream.on('data', (chunk: Buffer) => hasher.update(chunk))
+          stream.on('data', (chunk: string | Buffer) => hasher.update(chunk))
           stream.on('end', () => resolve(hasher.digest('hex')))
           stream.on('error', reject)
         })

--- a/src/engines/translator/GeminiTranslator.ts
+++ b/src/engines/translator/GeminiTranslator.ts
@@ -6,7 +6,21 @@ const DEFAULT_TIMEOUT_MS = 15_000
 
 const LANG_NAMES: Record<Language, string> = {
   ja: 'Japanese',
-  en: 'English'
+  en: 'English',
+  zh: 'Chinese',
+  ko: 'Korean',
+  fr: 'French',
+  de: 'German',
+  es: 'Spanish',
+  pt: 'Portuguese',
+  ru: 'Russian',
+  it: 'Italian',
+  nl: 'Dutch',
+  pl: 'Polish',
+  ar: 'Arabic',
+  th: 'Thai',
+  vi: 'Vietnamese',
+  id: 'Indonesian'
 }
 
 export class GeminiTranslator implements TranslatorEngine {

--- a/src/main/index.ts
+++ b/src/main/index.ts
@@ -31,7 +31,7 @@ import type { EngineConfig, TranslationResult } from '../engines/types'
 
 /** Scrub API keys from error messages before sending to renderer (#209) */
 function sanitizeErrorMessage(message: string): string {
-  const settings = store.store as Record<string, unknown>
+  const settings = store.store as unknown as Record<string, unknown>
   const secrets = [
     settings.googleApiKey,
     settings.deeplApiKey,
@@ -561,7 +561,7 @@ ipcMain.handle('get-settings', () => {
 
 // Subtitle settings — push changes to subtitle window in real-time
 ipcMain.handle('save-subtitle-settings', (_event, settings: Record<string, unknown>) => {
-  store.set('subtitleSettings', settings as import('./store').SubtitleSettings)
+  store.set('subtitleSettings', settings as unknown as import('./store').SubtitleSettings)
   subtitleWindow?.webContents.send('subtitle-settings-changed', settings)
 })
 

--- a/src/main/slm-worker.ts
+++ b/src/main/slm-worker.ts
@@ -247,7 +247,7 @@ async function handleTranslate(
 
     // Build prompt based on model type
     let prompt: string
-    let inferenceParams: { temperature: number; maxTokens: number; topK?: number; topP?: number; repeatPenalty?: number }
+    let inferenceParams: { temperature: number; maxTokens: number; topK?: number; topP?: number; repeatPenalty?: { penalty: number } }
 
     if (activeModelType === 'hunyuan-mt-15') {
       // HY-MT1.5 uses the official Tencent prompt template:
@@ -261,7 +261,7 @@ async function handleTranslate(
         prompt = `${contextSection}Translate the following segment into ${toLang}, without additional explanation.\n\n${text}`
       }
       // HY-MT1.5 recommended parameters (same as Hunyuan-MT)
-      inferenceParams = { temperature: 0.7, maxTokens: 512, topK: 20, topP: 0.6, repeatPenalty: 1.05 }
+      inferenceParams = { temperature: 0.7, maxTokens: 512, topK: 20, topP: 0.6, repeatPenalty: { penalty: 1.05 } }
     } else if (activeModelType === 'hunyuan-mt') {
       // Hunyuan-MT uses a specific prompt template:
       // Chinese ↔ Other: Chinese prompt; Other ↔ Other: English prompt
@@ -275,7 +275,7 @@ async function handleTranslate(
         prompt = `${contextSection}Translate the following segment into ${toLang}, without additional explanation.\n\n${text}`
       }
       // Hunyuan-MT recommended parameters
-      inferenceParams = { temperature: 0.7, maxTokens: 512, topK: 20, topP: 0.6, repeatPenalty: 1.05 }
+      inferenceParams = { temperature: 0.7, maxTokens: 512, topK: 20, topP: 0.6, repeatPenalty: { penalty: 1.05 } }
     } else {
       // TranslateGemma prompt
       const contextSection = buildContextPrompt(translateContext)
@@ -367,7 +367,7 @@ async function handleTranslateIncremental(
     // Use responsePrefix to force the model to continue from previous output
     // This implements prefix-constrained decoding for SimulMT consistency
     const inferenceParams = (activeModelType === 'hunyuan-mt' || activeModelType === 'hunyuan-mt-15')
-      ? { temperature: 0.7, maxTokens: 512, topK: 20, topP: 0.6, repeatPenalty: 1.05 }
+      ? { temperature: 0.7, maxTokens: 512, topK: 20, topP: 0.6, repeatPenalty: { penalty: 1.05 } }
       : { temperature: 0.1, maxTokens: 512 }
 
     const response = await session.prompt(prompt, {

--- a/src/preload/index.d.ts
+++ b/src/preload/index.d.ts
@@ -36,6 +36,12 @@ export interface ElectronAPI {
   getWhisperVariants: () => Promise<Array<{ key: string; label: string; description: string; filename: string; sizeMB: number; downloaded: boolean }>>
   getMoonshineVariants: () => Promise<Array<{ key: string; label: string; description: string; modelId: string; sizeMB: number; params: string }>>
   getPlatform: () => Promise<string>
+  saveGlossary: (terms: Array<{ source: string; target: string }>) => Promise<void>
+  isDraftModelAvailable: () => Promise<boolean>
+  wsAudioStart: (port?: number) => Promise<void>
+  wsAudioStop: () => Promise<void>
+  wsAudioGetStatus: () => Promise<{ running: boolean; connected: boolean; port: number | null }>
+  onWsAudioStatus: (callback: (status: { running: boolean; connected: boolean; port: number | null }) => void) => (() => void)
 }
 
 declare global {

--- a/src/renderer/App.tsx
+++ b/src/renderer/App.tsx
@@ -1,8 +1,8 @@
-import { useEffect, useState } from 'react'
+import React, { useEffect, useState } from 'react'
 import SubtitleOverlay from './components/SubtitleOverlay'
 import SettingsPanel from './components/SettingsPanel'
 
-function App(): JSX.Element {
+function App(): React.JSX.Element {
   const [isSubtitleMode, setIsSubtitleMode] = useState(false)
 
   useEffect(() => {

--- a/src/renderer/components/SettingsPanel.tsx
+++ b/src/renderer/components/SettingsPanel.tsx
@@ -1,4 +1,4 @@
-import { useCallback, useEffect, useRef, useState } from 'react'
+import React, { useCallback, useEffect, useRef, useState } from 'react'
 import { useAudioCapture } from '../hooks/useAudioCapture'
 
 /** Supported language codes — must match Language type in types.ts */
@@ -42,7 +42,7 @@ interface DisplayInfo {
   label: string
 }
 
-function SettingsPanel(): JSX.Element {
+function SettingsPanel(): React.JSX.Element {
   const [engineMode, setEngineMode] = useState<EngineMode>('offline-hybrid')
   const [gpuInfo, setGpuInfo] = useState<{ hasGpu: boolean; gpuNames: string[] } | null>(null)
   const [apiKey, setApiKey] = useState('')
@@ -1452,7 +1452,7 @@ function SettingsPanel(): JSX.Element {
   )
 }
 
-function Section({ label, children, role }: { label: string; children: React.ReactNode; role?: string }): JSX.Element {
+function Section({ label, children, role }: { label: string; children: React.ReactNode; role?: string }): React.JSX.Element {
   return (
     <section style={{ marginBottom: '18px' }} role={role} aria-label={label}>
       <label style={sectionLabelStyle}>{label}</label>

--- a/src/renderer/components/SubtitleOverlay.tsx
+++ b/src/renderer/components/SubtitleOverlay.tsx
@@ -1,4 +1,4 @@
-import { useEffect, useRef, useState } from 'react'
+import React, { useEffect, useRef, useState } from 'react'
 
 interface SubtitleLine {
   id: number
@@ -34,7 +34,7 @@ const FADE_DURATION_MS = 8000
 const INTERIM_LINE_ID = -1
 const DRAFT_LINE_ID = -2
 
-function SubtitleOverlay(): JSX.Element {
+function SubtitleOverlay(): React.JSX.Element {
   const [lines, setLines] = useState<SubtitleLine[]>([])
   const [config, setConfig] = useState<SubtitleConfig>(DEFAULT_CONFIG)
   const fadeTimerRef = useRef<ReturnType<typeof setInterval> | null>(null)

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -8,6 +8,8 @@
     "isolatedModules": true,
     "moduleDetection": "force"
   },
+  "files": [],
+  "include": [],
   "references": [
     { "path": "./tsconfig.node.json" },
     { "path": "./tsconfig.web.json" }

--- a/tsconfig.node.json
+++ b/tsconfig.node.json
@@ -9,5 +9,5 @@
     "declarationMap": true,
     "composite": true
   },
-  "include": ["src/main/**/*", "src/engines/**/*", "src/pipeline/**/*", "src/logger/**/*", "electron.vite.config.ts"]
+  "include": ["src/main/**/*", "src/preload/**/*", "src/engines/**/*", "src/pipeline/**/*", "src/logger/**/*", "electron.vite.config.ts"]
 }

--- a/tsconfig.web.json
+++ b/tsconfig.web.json
@@ -10,5 +10,5 @@
     "declarationMap": true,
     "composite": true
   },
-  "include": ["src/renderer/**/*"]
+  "include": ["src/renderer/**/*", "src/preload/index.d.ts"]
 }

